docs/general/developer-resources/sapphire-paratime/getting-started.md

```diff
@@ -0,0 +1,288 @@
+
+You'll want a script to run some methods on the contract.
+Pop open your favorite editor and jam out the following stanza:
+```javascript
+
+```
+Save it to `scripts/exercise-contract.js`.
+We'll use it in just a bit.
+
+Next, you can run the following and see the contract being deployed.
+```
+PRIVATE_KEY="0x..." truffle migrate --network emerald
+```
+Everything should be smooth sailing so far.
+
+Finally, run this line and observe some output.
+```
+> PRIVATE_KEY="0x..." truffle exec --network emerald scripts/exercise.contract.js`
+
+Sent some coins in 0xf415ab586ef1c6c61b84b3bd803ae322f375d1d3164aa8ac13c9ae83c698a002
+A Transfer(0x56e5F834F88F9f7631E9d6a43254e173478cE06a, 0x56e5F834F88F9f7631E9d6a43254e173478cE06a, 42) was emitted.
+The balance storage slot contains 0x2a.
+The contract now has balance: 42
+```
+
+Great!
+That'll be the baseline for our confidential deployment.
+Stay tuned...
+
+### Deploy to Sapphire (confidential EVM)
+
+#### Get some sROSE
+
+Now for the fun part.
+As for Emerald, we need to configure the Sapphire network and get some tokens.
+Hit up the one and only [Oasis Testnet faucet](https://faucet.testnet.oasis.dev/) and this time select "Sapphire".
+Submit the form and your way.
+
+#### Add the Sapphire Testnet to Truffle
+
+And another diff for your applying pleasure:
+
+```diff
+diff --git a/truffle-config.js b/truffle-config.js
+index 7af2f42..0cd9d36 100644
+--- a/truffle-config.js
++++ b/truffle-config.js
+@@ -58,6 +58,11 @@ module.exports = {
+         new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.emerald.oasis.dev"),
+       network_id: 0xa515,
+     },
++    sapphire: {
++      provider: () =>
++        new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.sapphire.oasis.dev"),
++      network_id: 0x5aff,
++    },
+   },
+
+   // Set default mocha options here, use special reporters etc.
+```
+
+#### Port to Sapphire
+
+Here's where things start to get interesting.
+We're going to add confidentiality to this starter project in exactly two lines of code.
+
+You'll need to grab the Sapphire compatibility library ([@oasisprotocol/sapphire-paratime](https://www.npmjs.com/package/@oasisprotocol/sapphire-paratime)), so make that happen by issuing a
+```
+pnpm add @oasisprotocol/sapphire-paratime # npm also works
+```
+
+So far so good.
+No import it by adding this line to the top of `truffle-config.js`:
+```
+const sapphire = require('@oasisprotocol/sapphire-paratime');
+```
+That's the first line of code.
+Here's the second:
+
+```diff
+diff --git a/truffle-config.js b/truffle-config.js
+index 0cd9d36..7db7cf8 100644
+--- a/truffle-config.js
++++ b/truffle-config.js
+@@ -60,7 +60,7 @@ module.exports = {
+     },
+     sapphire: {
+       provider: () =>
+-        new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.sapphire.oasis.dev"),
++        sapphire.wrap(new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.sapphire.oasis.dev")),
```

Comment:
Typescript says sapphire.wrap doesn't accept HDWalletProvider.
```sh
Argument of type 'HDWalletProvider' is not assignable to parameter of type 'UpstreamProvider'.
  Type 'HDWalletProvider' is not assignable to type '{ send: AsyncSend; }'.
    Types of property 'send' are incompatible.
      Type '(payload: JSONRPCRequestPayload, callback: (error: Error | null, response: JSONRPCResponsePayload) => void) => void' is not assignable to type 'AsyncSend'.
        Types of parameters 'payload' and 'args' are incompatible.
          Property 'jsonrpc' is missing in type 'Web3ReqArgs' but required in type 'JSONRPCRequestPayload'.
```

Reply:
what version of `sapphire-paratime` and `HDWalletProvider` are you using?

Reply:
Can you apply this diff and see if it works?
```diff
diff --git a/clients/js/src/compat.ts b/clients/js/src/compat.ts
index 3b49049..c357e8f 100644
--- a/clients/js/src/compat.ts
+++ b/clients/js/src/compat.ts
@@ -57,6 +57,7 @@ export type AsyncSend = (
 ) => void;

 export interface Web3ReqArgs {
+  readonly jsonrpc?: string;
   readonly id?: string | number;
   readonly method: string;
   readonly params?: any[];
 ```
 but why would TS be complaining about a field not existing on an object that doesn't request it?

Reply:
(that fix was attempted in https://github.com/oasisprotocol/sapphire-paratime/pull/46)